### PR TITLE
feat: big numbers responsiveness

### PR DESCRIPTION
**Pull request by yasmin-e**
Closes: #4167 

### Description:
When tiles that contain 'big number' charts are resized to be smaller, the 'big number' should also become smaller as right now, they look disproportionately large on the dashboards (and sometimes, they don't even fit the tile).

### CODE REVIEW

**Review by IrakliJani — CHANGES_REQUESTED** [review]
there is an extra commit, branch needs a rebase

also, I think it's not working

CleanShot 2023-02-02 at 11 47 30

**Comment by hamzahc1** [comment]
@yasmin-e This doesn't look like it's responsive at the moment - when I stretch or shrink the tile, the big number stays the same.  

https://user-images.githubusercontent.com/14341285/216295998-7af345f1-5faf-4e24-9a78-9b668bf4e09b.mov

**Comment by yasmin-e** [comment]
> @yasmin-e This doesn't look like it's responsive at the moment - when I stretch or shrink the tile, the big number stays the same.
> Screen.Recording.2023-02-02.at.10.09.37.mov

I just checked the PR env and ... 

https://user-images.githubusercontent.com/67699259/216316459-c3b2fc40-b92a-4148-8b54-d1707f9e9b7b.mov

**Comment by PriPatel** [comment]
I couldn't seem to view this change in the PR link which is why I was confused. But it turns out that the **project** you are viewing matters. I didn't know this, so apologies.

I've attached two screenshots. Both are from different Projects in the same PR link. (both the projects are called Jaffle  Shop 🤔). Screenshot 2023-02-02 at 16 32 34
Screenshot 2023-02-02 at 16 32 57
I can see that the responsiveness works in one of them, but doesn't in the other.

File: packages/frontend/src/components/DashboardTiles/TileBase/TileBase.styles.tsx [hunk 1]
```diff
@@ -42,6 +43,8 @@ export const HeaderContainer = styled.div<HeaderContainerProps>`
                 }
             `
             : ''}
+    ${(props) =>
+        !props.hasTitle ? `margin-top: -5px; margin-bottom: -5px;` : ''}
```
Comment (IrakliJani) [inline packages/frontend/src/components/DashboardTiles/TileBase/TileBase.styles.tsx:47]: do we still need this? not sure what does this fix actually

File: packages/frontend/src/components/DashboardTiles/TileBase/index.tsx [hunk 2]
```diff
@@ -59,16 +59,20 @@ const TileBase = <T extends Dashboard['tiles'][number]>({
...
             className={isLoading ? Classes.SKELETON : undefined}
             isEditMode={isEditMode}
             isHovering={isHovering}
+            hasTitle={hasTitle}
         >
             <HeaderContainer
                 isEditMode={isEditMode}
                 onMouseEnter={() => setIsHovering(true)}
                 onMouseLeave={() => setIsHovering(false)}
+                hasTitle={hasTitle}
```
Comment (IrakliJani) [inline packages/frontend/src/components/DashboardTiles/TileBase/index.tsx:75]: you can just `!hideTitle`
was little bit confusing to see `hideTitle ? false : true`

File: packages/frontend/src/components/SimpleStatistic/SimpleStatistics.styles.tsx [hunk 3]
```diff
@@ -16,10 +17,11 @@ interface BigNumberProps extends React.HTMLAttributes<HTMLDivElement> {
 
 export const BigNumber = styled.div<BigNumberProps>`
     ${({ $interactive }) => ($interactive ? 'cursor: pointer;' : '')}
-    font-size: 4em;
+    font-size: 1em;
```
Comment (IrakliJani) [inline packages/frontend/src/components/SimpleStatistic/SimpleStatistics.styles.tsx:20]: can we get rid of em-s here and use pixels?

File: packages/frontend/src/components/SimpleStatistic/SimpleStatistics.styles.tsx [hunk 4]
```diff
@@ -2,6 +2,7 @@ import { Colors } from '@blueprintjs/core';
 import styled from 'styled-components';
 
 export const BigNumberContainer = styled.div`
+    margin-top: 2px;
```
Comment (IrakliJani) [inline packages/frontend/src/components/SimpleStatistic/SimpleStatistics.styles.tsx:5]: does this solve anything?

**Review by IrakliJani — CHANGES_REQUESTED** [review]
few comments. and one bug which is about large tiles

we should also have max font sizes for both the label and the big number

here is a problem in the explore page

CleanShot 2023-02-02 at 20 14 34@2x

and this one is less likely but still weird for dashboard tile with a full screen width

CleanShot 2023-02-02 at 20 18 09@2x

File: packages/frontend/src/components/DashboardTiles/TileBase/TileBase.styles.tsx (hunk 1 above)
Comment (yasmin-e) [reply packages/frontend/src/components/DashboardTiles/TileBase/TileBase.styles.tsx:47]: it just makes the header more compact when there's no title

**Review by IrakliJani — APPROVED** [review]
looks good!

**Pull request merged**